packages/labs/testing/src/lib/worker.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {parentPort, workerData} from 'worker_threads';
+import {render} from '@lit-labs/ssr/lib/render-with-global-dom-shim.js';
+
+import type {Payload} from './lit-ssr-plugin.js';
+
+if (parentPort === null) {
+  throw new Error('worker.js must only be run in a worker thread');
+}
+
+const {template, modules} = workerData as Payload;
+
+await Promise.all(modules.map((module) => import(module)));
+
+// Dangerously spoof TemplateStringsArray by adding back the `raw` property
+// property which gets stripped during serialization of the TemplateResult.
+// This is needed to get through the check here
+// https://github.com/lit/lit/blob/1fbd2b7a1e6da09912f5c681d2b6eaf1c4920bb4/packages/lit-html/src/lit-html.ts#L867
+const strings = template.strings;
+(strings as {raw: TemplateStringsArray['raw']}).raw = strings;
+
+let rendered = '';
+for (const str of render({...template, strings}, {deferHydration: true})) {
```

Comment:
Not familiar with how workers work... since this is all executed in module scope, if it throws, then we'll get a worker.on('error') callback?

Reply:
Yes that's correct. Currently the wtr server command simply fails for those cases so the fixture function call will then throw. We see that happen in the bad-element test.